components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec

```diff
@@ -0,0 +1,191 @@
+# build/install configurations, and supply additional files to support
+# desired integration conventions.
+#
+#----------------------------------------------------------------------------eh-
+
+%include %{_sourcedir}/OHPC_macros
+
+%global gnuver 6.3.0
+%global cmakemajor 3.8
+%global cmakever 3.8.1
+
+%define pname llvm-compilers
+
+Summary:   The LLVM Compiler Infrastructure
+Name:      %{pname}%{PROJ_DELIM}
+Version:   4.0.0
+Release:   1
+License:   UIUC
+Group:     %{PROJ_NAME}/compiler-families
+URL:       http://www.llvm.org
+BuildRoot: %{_tmppath}/%{pname}-%{version}-%{release}-root
+Source0:   http://releases.llvm.org/%{version}/llvm-%{version}.src.tar.xz
+Source1:   http://releases.llvm.org/%{version}/cfe-%{version}.src.tar.xz
+Source2:   http://releases.llvm.org/%{version}/compiler-rt-%{version}.src.tar.xz
+Source3:   http://releases.llvm.org/%{version}/libcxx-%{version}.src.tar.xz
+Source4:   http://releases.llvm.org/%{version}/libcxxabi-%{version}.src.tar.xz
+Source5:   http://releases.llvm.org/%{version}/libunwind-%{version}.src.tar.xz
+Source6:   http://releases.llvm.org/%{version}/lld-%{version}.src.tar.xz
+Source7:   http://releases.llvm.org/%{version}/openmp-%{version}.src.tar.xz
+Source8:   https://cmake.org/files/v%{cmakemajor}/cmake-%{cmakever}.tar.gz
+Source9:   OHPC_macros
+Source10:  OHPC_setup_compiler
+Patch1:    openmp-4.0.0-gsupport.patch
+
+%define debug_package %{nil}
+
+BuildRequires: gnu-compilers%{PROJ_DELIM}
+BuildRequires: make
+BuildRequires: perl
+BuildRequires: python
+BuildRequires: pkgconfig
+BuildRequires: binutils-devel
+Requires:      binutils
+Requires:      gnu-compilers%{PROJ_DELIM} = %{gnuver}
+%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel}
+BuildRequires: perl-Data-Dumper
+%endif
+
+%define install_path %{OHPC_COMPILERS}/llvm/%{version}
+%define gnu_path %{OHPC_COMPILERS}/gcc/%{gnuver}
+%if 0%{?sles_version} || 0%{?suse_version}
+%define compiler_path `%{OHPC_COMPILERS}/gcc/%{gnuver}/bin/g++ -v --version 2>&1|grep COMPILER_PATH|cut -f 2- -d '='`
+%define library_path `%{OHPC_COMPILERS}/gcc/%{gnuver}/bin/g++ -v --version 2>&1|grep LIBRARY_PATH|cut -f 2- -d '='`
+%endif
+
+%description
+
+The LLVM Compiler Infrastructure.
+
+%prep
+%setup -q -c -b1 -b2 -b3 -b4 -b5 -b6 -b7 -b8
+%patch1 -p0
+
+cd llvm-%{version}.src/tools
+ln -s ../../cfe-%{version}.src clang
+cd ../..
+
+cd llvm-%{version}.src/tools
+ln -s ../../lld-%{version}.src lld
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../compiler-rt-%{version}.src compiler-rt
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libcxx-%{version}.src libcxx
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libcxxabi-%{version}.src libcxxabi
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libunwind-%{version}.src libunwind
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../openmp-%{version}.src openmp
+cd ../..
+
+ln -s llvm-%{version}.src llvm
+ln -s cmake-%{cmakever} cmake
+
+%build
+
```

Comment:
For some reason many of the OpenHPC packages do it that way. I just followed the order that I encountered.